### DISCUSSION TITLE
Illustration and tables list for docx document

### DISCUSSION THREAD

**Discussion by Pierre9344** [quarto-dev/quarto-cli#8503] [Feature Requests]
### Description

Hello,

Would it be possible for quarto to automatically create lists of tables when rendering to docx document? This already exist for the pdf output (option lot and lof).

Thank you in advance for your help

**Comment by cderv** [comment]
For reference, `lot` and `lof` for PDF are brought by Pandoc directly. There are feature requests there for same feature in DOCX: 
* https://github.com/jgm/pandoc/issues/1965
* https://github.com/jgm/pandoc/issues/8245

This was also already discussed here at: 

* https://github.com/quarto-dev/quarto-cli/discussions/5945

And this is tracked already in: 
* https://github.com/quarto-dev/quarto-cli/issues/2138

It did not make it to 1.4 but we marked it for 1.5. You should follow there.

**Comment by rgaiacs** [comment]
> Would it be possible for quarto to automatically create lists of tables when rendering to docx document?

As mentioned in https://github.com/jgm/pandoc/issues/7451#issuecomment-899116446, it is possible to use DOCX native table numbering. For example, `pandoc --from markdown --to docx+native_numbering --output 8503-quarto.docx mwe.md` where `mwe.md` is

```
---
title: MWE
---

Lorem ipsum.

---
A B
- -
1 2
3 4
---

Table: Here's the caption.
```

will produce [8503-quarto.docx](https://github.com/quarto-dev/quarto-cli/files/14095039/8503-quarto.docx)

![Screenshot 2024-01-30 094333](https://github.com/quarto-dev/quarto-cli/assets/1506457/45366f9f-7e63-4815-a1b8-35e412ffb895)

Manually clicking in `Insert table of Figures` will change the document to

![Screenshot 2024-01-30 094631](https://github.com/quarto-dev/quarto-cli/assets/1506457/c8bb04da-d3ce-4891-9362-d65ea9f9af3c)

**Reply by cderv** [reply]
Good to know ! Thanks for sharing ! 

Just a warning that this need to be checked again cross-ref feature - when using [cross reference](https://quarto.org/docs/authoring/cross-references.html) syntax, this trick may not work anymore. 

That is what https://github.com/quarto-dev/quarto-cli/issues/2138 is about as part of 
* https://github.com/quarto-dev/quarto-cli/issues/4944.

**Reply by mweylandt** [reply]
Interestingly, this seems to work inconsistently/differentially for figures and tables based on whether I'm rendering from the command line or from RStudio. If I render the following doc to .md and convert using the command line command above, the figures work and not the tables -- and it breaks cross-references. (in quarto 1.3 & 1.4). If on the other hand I specify docx+native_numbering in the quarto YAML, the tables work (though with repetitive names) and the figures don't. (in quarto 1.3, but not in 1.4)

```
---
title: "Table test"
format: docx+native_numbering
keep-md: true
execute: 
  message: false
  warning: false
  echo: false
---

Cross-references:
- figure: @fig-test1
- table: @tbl-test2

```{r}
#| fig-cap: "test"
#| label: fig-test1
library(tidyverse)
library(knitr)

mtcars |>  ggplot(aes(x = as.factor(am), y = mpg)) + geom_boxplot()
`` `

Table: 
```{r}
#| label: tbl-test2
#| tbl-cap: "test table labels"
#| echo: false
iris |> select(1:4) |> head() |> kable()
`` `


```

**Reply by mcanouil** [reply]
The cross-reference system changed completely between 1.3 and 1.4, especially to allow custom cross-reference.

**Reply by cderv** [reply]
It is possible our cross-reference system does not work with DOCX own numbering feature activated through the pandoc extension.

`native_numbering` is deactivated by default, and I don't think we tested against its activation. So for now, I would say cross referencing feature of Quarto is not compatible with `docx+native_numbering`

You could open an issue so that we can see if and how we can adapt. But it could be hard.

**Reply by mweylandt** [reply]
Just wanted to let people looking at this thread know that this workaround has effects on cross references (across both systems, 1.3 and 1.4, hence me specifying both) 😊

Thank you! I think I'll hold off on an issue for now, as I understand from reading
issues and roadmaps the list of tables issue is on the to-do list anyway :)
appreciate the work y'all do!

**Comment by mweylandt** [comment]
Thank you! I think I'll hold off for now, as I understand from reading
issues and roadmaps the list of tables issue is on the Todo list anyway :)
appreciate the work y'all do!
